tests/post/steps/test_monitoring.py

```diff
@@ -0,0 +1,103 @@
+import requests
+import json
+
+from pytest_bdd import scenario, given, then, parsers
+
+from tests import kube_utils
+from tests import utils
+
+
+# Scenarios {{{
+
+@scenario('../features/monitoring.feature', 'List Pods')
+def test_list_pods(host):
+    pass
+
+
+@scenario('../features/monitoring.feature', 'Expected Pods')
+def test_expected_pods(host):
+    pass
+
+
+@scenario('../features/monitoring.feature', 'Monitored components statuses')
+def test_monitored_components(host):
+    pass
+
+
+# }}}
+# Given {{{
+
+@given("the Prometheus API is available")
```

Comment:
I think the [Given](https://cucumber.io/docs/gherkin/reference/#given) are used to modify the "environment" to be in a desired state
The code here looks like a [Then](https://cucumber.io/docs/gherkin/reference/#then) to me, since we are checking a state (even the function is called `check`, that's a sign).

Reply:
True, I will change it.

Reply:
Done.

Reply:
> I think the [Given](https://cucumber.io/docs/gherkin/reference/#given) are used to modify the "environment" to be in a desired state

I kind of disagree here: we have the `Given the K8s API is available` in a lot of other scenarii. To me, `Given`s are about making sure your environment is in the expected state **before** applying some actions, that live in `When`s. If the `Given` needs to change something for matching the desired state, fine, but it's not required IMHO.

> The code here looks like a [Then](https://cucumber.io/docs/gherkin/reference/#then) to me, since we are checking a state (even the function is called `check`, that's a sign).

It would be a `Then` if it's a result of a `When` action that we want to check.